In [3]:
import numpy as np

def rule3 (da, db):
    dQ = np.sqrt(da**2+db**2)
    return dQ

d1 = 1.0
d2 = 0.3
errd1 = 0.0005
errd2 = 0.0005
dtotal = d1 + d2
errdtotal = rule3(errd1, errd2)
print("dtotal =", dtotal)
print("errdtotal =", errdtotal)


dtotal = 1.3
errdtotal = 0.0007071067811865475


In [4]:
def rule1 (c,da):
    dQ=(c*da)
    return dQ

c = 2
da = 3
dQ = rule1(c,da)
print (dQ)

6


In [6]:
def rule2 (c,A,m,da):
    dQ=(c*m*A**(m-1)*da)
    return dQ

c = 2
A = 3
m = 2
da = 3
dQ = rule2 (c,A,m,da)
print (dQ)

36


In [3]:
def rule4 (Q,m,n,da,a,db,b):
    dQ= Q*np.sqrt(((m*da)/a)**2 + ((n*db)/b)**2)
    return dQ

Q = 1
m = 2
n = 3
da = 4
a = 5
db = 6
b = 7
dQ = rule4 (Q,m,n,da,a,db,b)
print (dQ)   

3.028571428571429


In [4]:
def rule4 (v,a,dM,M,b,dm,m,c,dR,R,d,dh,h):
    dv= v*np.sqrt(((a*dM)/M)**2 + ((b*dm)/m)**2 + ((c*dR)/R)**2 + ((d*dh)/h)**2)
    return dv

v = 2.384
a = 1
dM = .0001
M = .2578
b = -1
dm = .0001
m = .0659
c = .5
dR = .005
R = .28
d = .5
dh = .000516
h = .06768
dv = rule4 (v,a,dM,M,b,dm,m,c,dR,R,d,dh,h)
print (dv)

0.023443859565717474


In [2]:
x = np.array ([1.1,1.3,1.4,0.9,0.95,1.05])
xmean = np.average(x)
xstdev = np.std(x)
print (xmean)
print (xstdev)

1.1166666666666667
0.17950549357115014


The average is 1.12 and the standard deviation is .18

$\delta Q = \sqrt{\delta A^2 + \delta B^2} $

$\delta v = v \sqrt{(\frac{1\delta x}{x})^2 + (\frac{-.5\delta y}{y})^2}$